processor/src/main/java/org/mapstruct/ap/internal/model/source/TargetReference.java

```diff
@@ -246,9 +247,22 @@ else if ( targetWriteAccessor == null ) {
             }
         }
 
-        private void reportMappingError(Message msg, Object... objects) {
-            messager.printMessage( method.getExecutable(), mapping.getMirror(), mapping.getSourceAnnotationValue(),
-                msg, objects );
+        private void reportMappingError() {
+            if ( errorMessage == Message.BEANMAPPING_UNKNOWN_PROPERTY_IN_RESULTTYPE ) {
```

Comment:
Actually I did not quite like the if-statement itself. It's like we set the error message depending on some conditions in one place (`setErrorMessage()`), but then we again check on error condition message to set the correct arguments in another place (`reportMappingError()`). In theory we could create some ErrorMessage object with the error message with argument list in `setErrorMessage()` or even create a strategy of reporting the error, but I think that in this case it would be an overcomplication. If we can live with this `if` statement we may leave it as is.